chain/transaction.go

```diff
@@ -376,41 +328,22 @@ func (t *Transaction) Execute(
 // Sponsor is the [codec.Address] that pays fees for this transaction.
 func (t *Transaction) GetSponsor() codec.Address { return t.Auth.Sponsor() }
 
-func (t *Transaction) Marshal(p *codec.Packer) error {
-	if len(t.bytes) > 0 {
-		p.PackFixedBytes(t.bytes)
-		return p.Err()
-	}
-	return t.marshal(p)
-}
-
 type txJSON struct {
-	ID      ids.ID      `json:"id"`
-	Actions codec.Bytes `json:"actions"`
-	Auth    codec.Bytes `json:"auth"`
-	Base    *Base       `json:"base"`
+	ID      ids.ID        `json:"id"`
```

Comment:
Should we export this type for use in the APIs? We cannot marshal and unmarshal a transaction as JSON, we require unmarshalling it with the parser, which is non-standard.

It would be better to export the intermediate type and use that for JSON marshal/unmarshal cc @tsachiherman